akka-stream/src/main/scala/akka/stream/impl/io/FilePublisher.scala

```diff
@@ -52,6 +52,7 @@ private[akka] final class FilePublisher(f: Path, completionPromise: Promise[IORe
       chan = FileChannel.open(f, FilePublisher.Read)
     } catch {
       case ex: Exception ⇒
+        completionPromise.success(IOResult(0L, Failure(ex)))
```

Comment:
Boyscouting is always in scope :-)

-- 
Cheers,
√
On May 26, 2016 9:31 AM, "Heiko Seeberger" <notifications@github.com> wrote:

> In akka-stream/src/main/scala/akka/stream/impl/io/FilePublisher.scala
> <https://github.com/akka/akka/pull/20620#discussion_r64701885>:
>
> > @@ -52,6 +52,7 @@ private[akka] final class FilePublisher(f: Path, completionPromise: Promise[IORe
> >        chan = FileChannel.open(f, FilePublisher.Read)
> >      } catch {
> >        case ex: Exception ⇒
> > +        completionPromise.success(IOResult(0L, Failure(ex)))
>
> @patriknw <https://github.com/patriknw>, that will only be executed if
> the channel is not null, so there is zero risk for a collision.
> @viktorklang <https://github.com/viktorklang>, I agree, but that's out of
> scope for this PR.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/akka/akka/pull/20620/files/affe3e43b92d1a55e05520b58802bd4181ca65a4#r64701885>
>